core/src/main/java/hudson/util/PluginServletFilter.java

```diff
@@ -142,5 +144,25 @@ public void destroy() {
         list.clear();
     }
 
+    @Restricted(NoExternalUse.class)
+    public static void cleanUp() {
+        PluginServletFilter instance = getInstance(Jenkins.getInstance().servletContext);
+        if (instance != null) {
+            for (Iterator<Filter> iterator = instance.list.iterator(); iterator.hasNext(); ) {
+                Filter f = iterator.next();
+                try {
+                    f.destroy();
+                } catch (RuntimeException e) {
+                    LOGGER.log(Level.WARNING, "Filter " + f + " propagated an exception from its destroy method", e);
+                } catch (Error e) {
+                    throw e; // we are not supposed to catch errors, don't log as could be an OOM
```

Comment:
Yes you have a valid point, catching `LinkageError` is probably one of the few errors to catch... and while theoretically possible (e.g. the `destroy` method references a method that does not exist) I think for this case it is exceedingly unlikely